File: SpringBoot_Backend_Exam_portal/src/main/java/com/exam/services/QuestionService.java
package com.exam.services;

import java.util.Set;

import com.exam.entity.Question;
import com.exam.entity.Quiz;

public interface QuestionService {
	public Question addQuestion(Question question);
	public Question updateQuestion(Question question);
	public Set<Question> getQuestions();
	public Question getQuestion(Long questionId);
	public Set<Question> getQuestionOfQuiz(Quiz quiz);
	public void deleteQuestion(Long quesId);
	public Question get(Long questionsId);
 
}

File: Angular_Exam_Portal/app/services/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseurl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  constructor(private http:HttpClient) { }
  public getQuestionOfQuiz(qid:any){
    return this.http.get(`${baseurl}/question/quiz/all/${qid}`);
  }

  public addQuestion(question:any){
    return this.http.post(`${baseurl}/question/`,question);
  }

  public deleteQuestion(questionId:any){
    return this.http.delete(`${baseurl}/question/${questionId}`);
  }
  public getQuestionOfQuizForTest(qid:any){
    return this.http.get(`${baseurl}/question/quiz/${qid}`);
  }

  public evalQuiz(questions:any){
    return this.http.post(`${baseurl}/question/eval-quiz`,questions)
  }

}
